src/main/java/develhope/Flow_Open_Spring/controller/ProductController.java

```diff
@@ -27,26 +29,32 @@ public List<Product> getProduct() {
     }
 
     @GetMapping("/{id}")
-    public Optional<Product> getOneProduct(@PathVariable Long id) {
-        return productRepository.findById(id);
+    public ResponseEntity getOneProduct(@PathVariable Long id) {
+        Optional<Product> findProduct = productRepository.findById(id);
+        if (findProduct.isPresent()) {
+            productRepository.findById(id);
+            ResponseEntity.status(HttpStatus.OK);
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Did not find product!");
```

Comment:
dopo l'if metti un else not found. 
la bad request invece va bene lì, però se riesci mettila a una condizione tipo di sintassi errata, sennò va bene così

Reply:
confermo meglio l'errore corretto NOTFOUND la BADREQUEST aggiungila dopo come statement finale